pam/pam_adsys.c

```diff
@@ -223,14 +262,36 @@ static int set_dconf_profile(pam_handle_t * pamh, const char *username,
 {
 	int retval = PAM_SUCCESS;
 
+	char* profile_name = (char*)malloc((strlen(username) + 1) * sizeof(char));
+	strcpy(profile_name, username);
+
+	// We need to check if the profile name does not already contain the domain or is in the \\format
+	if (strchr(profile_name, '@') == NULL && strchr(profile_name, '\\') == NULL) {
+		char* domain = get_default_domain(pamh);
```

Comment:
Also, I don’t see `domain` being freed?

Reply:
Once I `strcat` the `domain` into `profile_name`, when I free `profile_name` doesn't it free the `domain` as well?

Reply:
```
       strcat()
              This  function catenates the string pointed to by src, after the
              string pointed to  by  dst  (overwriting  its  terminating  null
              byte).   The programmer is responsible for allocating a destina‐
              tion buffer large enough, that is, strlen(dst) +  strlen(src)  +
              1.
```.

So, this means that the new allocated has nothing to do to src and dst, and so, if we allocated any of them, we need to free them.